src/script/components/manifest-previewer/manifest-previewer.ts

```diff
@@ -0,0 +1,559 @@
+
+    .buttons-div {
+      display: flex;
+      justify-content: space-between;
+      margin: 71px auto 0;
+      width: 272px;
+    }
+
+    fast-button.platform-button {
+      height: 35px;
+      border-radius: 33px;
+      font-size: 12.5751px;
+      line-height: 19px;
+      width: 80px;
+      background: #FFF;
+      box-shadow: 0px 3px 3.02588px rgba(0, 0, 0, 0.25);
+      color: var(--font-color);
+    }
+    
+    fast-button::part(control) {
+      font-weight: 700;
+    }
+
+    .platform-button.selected {
+      background: #292C3A;
+      box-shadow: 0px 0.75647px 3.02588px rgba(0, 0, 0, 0.25);
+      color: #FFF;
+    }
+
+    .name {
+      background: rgba(194, 194, 194, 0.4);
+      border-radius: 4px;
+      height: 24px;
+      font-weight: 700;
+      font-size: 16px;
+      line-height: 25px;
+      text-align: center;
+      color: #000;
+      margin: 20px auto 0;
+      width: fit-content;
+      padding: 0 5px;
+    }
+
+    .preview-text {
+      position: absolute;
+      bottom: 25px;
+      left: calc(50% - 55px);
+      font-weight: 400;
+      font-size: 10px;
+      line-height: 16px;
+      text-align: center;
+      color: var(--secondary-font-color);
+      width: 110px;
+    }
+
+    img.nav-arrow-right {
+      position: absolute;
+      width: 19px;
+      height: 38px;
+      top: 377px;
+      right: 16px;
+      cursor: pointer;
+    }
+    
+    img.nav-arrow-left {
+      position: absolute;
+      width: 19px;
+      height: 38px;
+      top: 377px;
+      left: 16px;
+      transform: rotate(180deg);
+      cursor: pointer;
+    }
+
+    .screen-title {
+      margin: 10px auto;
+      width: fit-content;
+      font-weight: 600;
+      font-size: 14px;
+      text-align: center;
+    }
+
+    .screen-info {
+      margin: 0 auto;
+      font-weight: 400;
+      font-size: 12px;
+      line-height: 16px;
+      text-align: center;
+      color: var(--secondary-font-color);
+      width: 230px;
+      display: block;
+    }
+
+    /* The card is hidden for smaller screens */
+    @media(min-width: 800px) {
+      .card {
+        display: block;
+      }
+    }
+
+    /* 800 designs */
+    @media(min-width: 800px) and (max-width: 1024px) {
+      .card {
+        width: 354px;
+      }
+    }
+
+    /* 1024 designs */
+    @media(min-width: 1024px) and (max-width: 1366px) {
+      .card {
+        width: 366px;
+      }
+    }
+
+    /* 1366 designs */
+    @media(min-width: 1366px) {
+      .card {
+        width: 479.03px;
+      }
+
+      img.nav-arrow-right {
+        right: 30px;
+      }
+
+      img.nav-arrow-left {
+        left: 30px;
+      }
+    }
+  `;
+
+  /**
+   * The website's URL.
+   */
+  @property() private siteUrl = '';
+  
+  /**
+   * The current preview screen.
+   */
+  @property() stage: PreviewStage = 'name';
+  
+  /**
+   * The input web manifest.
+   */
+  @property({ 
+    type: Object,
+    converter: value => {
+      if (!value) {
+        return undefined;
```

Comment:
this seems like an edge case we can guard against in the parent, what are you seeing here? The default converter is going to be JSON.parse for the `Object` type